client/history/peerResolver.go

```diff
@@ -29,15 +29,25 @@ import (
 // PeerResolver is used to resolve history peers.
 // Those are deployed instances of Cadence history services that participate in the cluster ring.
 // The resulting peer is simply an address of form ip:port where RPC calls can be routed to.
-type PeerResolver struct {
+//
+//go:generate mockgen -package $GOPACKAGE -source $GOFILE -destination peerResolver_mock.go -package history github.com/uber/cadence/client/history PeerResolver
+type PeerResolver interface {
+	FromWorkflowID(workflowID string) (string, error)
+	FromDomainID(domainID string) (string, error)
+	FromShardID(shardID int) (string, error)
+	FromHostAddress(hostAddress string) (string, error)
+	GetAllPeers() ([]string, error)
+}
+
+type peerResolver struct {
 	numberOfShards int
 	resolver       membership.Resolver
 	namedPort      string // grpc or tchannel, depends on yarpc configuration
 }
 
 // NewPeerResolver creates a new history peer resolver.
 func NewPeerResolver(numberOfShards int, resolver membership.Resolver, namedPort string) PeerResolver {
-	return PeerResolver{
+	return peerResolver{
```

Comment:
let's change this to `&peerResolver` and attach the functions to the `*peerResolver`